nemoguardrails/llm/default_framework.py

```
@@ -0,0 +1,138 @@
+    async def reset(self) -> None:
+        """Close all pooled HTTP clients and clear registered providers.
+
+        Destructive: clears both the client pool and any providers registered
+        via ``register_provider``. Callers expecting custom providers to
+        survive must re-register them after ``reset``.
+        """
+        for client in list(self._clients.values()):
+            try:
+                await client.close()
+            except Exception:
+                pass
```

Ruff (0.15.11)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)